optimus.py

```diff
@@ -0,0 +1,798 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+__license__ = "GPL-2.0"
+__author__ = "Martin Kurbanov"
+__email__ = "mmkurbanov@salutedevices.com"
+__copyright__ = "Copyright (c) 2023, SaluteDevices"
+__version__ = '0.0.1'
+
+try:
+    from pyamlboot.pyamlboot import pyamlboot
+except ImportError:
+    from pyamlboot import pyamlboot
+
+import logging
+import time
+import typing
+from collections import OrderedDict
+from dataclasses import dataclass
+from struct import pack, unpack
+
+import usb_backend
+
+USB_BACKEND = usb_backend.get_backend()
+
+
+class BulkCmdError(Exception):
+    pass
+
+
+class TplCmdError(Exception):
+    pass
+
+
+class SocId:
+    STAGE_MINOR_IPL = 0   # Initial Program Loader
+    STAGE_MINOR_SPL = 8   # Secondary Program Loader
+    STAGE_MINOR_TPL = 16  # Tertiary Program Loader
+
+    def __init__(self, socid):
```

Comment:
To avoid confusion I'd probably rename the `socid` parameter to `rawData` or `data` or something similar, just like the local field `_raw`.